packages/project/src/db/tree.ts

```diff
@@ -1,12 +1,126 @@
-import { type Tree, Instance } from "@webstudio-is/react-sdk";
+import { formatAsset } from "@webstudio-is/asset-uploader/server";
+import { type Tree, Instance, Text } from "@webstudio-is/react-sdk";
 import { applyPatches, type Patch } from "immer";
 import {
+  Asset,
   prisma,
   type Prisma,
   type Tree as DbTree,
 } from "@webstudio-is/prisma-client";
 import { utils } from "../index";
-import type { Breakpoint } from "@webstudio-is/css-data";
+import {
+  type Breakpoint,
+  SharedStyleValue,
+  ImageValue,
+} from "@webstudio-is/css-data";
+import { z } from "zod";
+import DataLoader from "dataloader";
+import warnOnce from "warn-once";
+
+const assetsLoader = new DataLoader<string, Asset | undefined>(
```

Comment:
I am missing something, what is DataLoader helping us with as opposed to e.g. just a function that does the same thing?